### PR TITLE
Fix merge scheduler config settings

### PR DESCRIPTION
**Pull request by jimczi**
This change fixes the retrieval of the merge scheduler config that previously used IndexSettings#getValue and failed to
retrieve the current value of the max_thread_count when max_merge_count was set.
Instead we now use Setting#get(Settings) which correctly retrieves the value (and default value) for all settings.
This fixes a bug that appears when only max_thread_count was updated. Any value for the setting was accepted leading to a state where the
 merge scheduler throws exceptions (and fails any merge) and the index cannot be opened anymore (if it was closed before).

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/MergeSchedulerConfig.java [hunk 1]
```diff
@@ -69,8 +69,9 @@
     private volatile int maxMergeCount;
 
     MergeSchedulerConfig(IndexSettings indexSettings) {
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/index/MergeSchedulerConfig.java]: Can this class be unit tested instead of depending on integration tests?

**Review by rjernst — COMMENTED** [review]
I'm confused on what the actual issue was here.  The only difference between what you have done here, and what existed with `getValue()` was using the raw `settings` in `IndexSettings`, vs the `scopedSettings`. And the latter is what is still used for updating. Is this actually a problem in `AbstractScopedSettings`?

**Comment by jbbarth** [comment]
@rjernst if it helps reproducing the problem, I think this PR aims at fixing a problem I reported on the forum here: https://discuss.elastic.co/t/problem-updating-settings-maxthreadcount-8-should-be-maxmergecount-6/76279

File: core/src/main/java/org/elasticsearch/common/settings/Setting.java [hunk 2]
```diff
@@ -378,7 +377,7 @@ public final T get(Settings primary, Settings secondary) {
             return get(primary);
         }
         if (fallbackSetting == null) {
-            return get(secondary);
+            return exists(secondary) ? get(secondary) : get(primary);
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/common/settings/Setting.java]: I think this block and the following one could be reordered to get the same effect?
```
if (exists(secondary)) {
    return get(secondary);
}
if (fallbackSetting == null) {
    return get(primary);
}

**Review by rjernst — APPROVED** [review]
Thanks for the new tests!! And also for tracking down the root cause. I have one suggestion.

**Comment by jimczi** [comment]
Thanks @rjernst and @jbbarth !

**Pull request merged**